feat(parameters): migrate AppConfig to new APIs due to API deprecation (#1553)
awslabs/aws-lambda-powertools-python: docs/utilities/feature_flags.md

description: Utility
---

The feature flags utility provides a simple rule engine to define when one or multiple features should be enabled depending on the input.

???+ info
    We currently only support AppConfig using [freeform configuration profile](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-creating-configuration-and-profile.html#appconfig-creating-configuration-and-profile-free-form-configurations).

>     We are planning to add support for the [feature flag configuration profile](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-creating-configuration-and-profile.html#appconfig-creating-configuration-and-profile-feature-flags) in future releases.

Reviewer: ```suggestion
```

In this case, it's best to wait for customer demand via feature request (well known format). If we add that we're planning to add support now, we'll accidentally commit and prioritize over existing backlog items other customers asked.

* [Feature Toggles (aka Feature Flags) - Pete Hodgson](https://martinfowler.com/articles/feature-toggles.html)
* [AWS Lambda Feature Toggles Made Simple - Ran Isenberg](https://isenberg-ran.medium.com/aws-lambda-feature-toggles-made-simple-580b0c444233)
* [Feature Flags Getting Started - CloudBees](https://www.cloudbees.com/blog/ultimate-feature-flag-guide)

> * [Best Practices for validating AWS AppConfig Feature Flags and Configuration Data](https://aws.amazon.com/pt/blogs/mt/best-practices-for-validating-aws-appconfig-feature-flags-and-configuration-data/)

Reviewer: ```suggestion
```

Not applicable here. You also accidentally shared a `pt` domain

???+ note

>     Optimize your Lambda execution time by putting your feature settings in a single [Configuration](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-creating-configuration-and-profile.html).

Reviewer: ```suggestion
    AWS AppConfig requires two API calls to fetch configuration for the first time. You can improve latency  by consolidating your feature settings in a single [Configuration](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-creating-configuration-and-profile.html).
```

In this case, a tip is to share why they should continue reading this note - Challenge first, recommended step afterward. If the challenge doesn't concern them, they can stop reading immediately. This optimizes for cognitive load.

### IAM Permissions

> Your Lambda function must have `appconfig:GetLatestConfiguration` and `appconfig:StartConfigurationSession` IAM permissions in order to fetch configuration from AWS AppConfig.

Reviewer: ```suggestion
Your Lambda function IAM Role must have `appconfig:GetLatestConfiguration` and `appconfig:StartConfigurationSession` IAM permissions before using this feature.
```

fixing older wording
